src/zkbob/ZkBobCompoundingMixin.sol

```diff
@@ -0,0 +1,194 @@
+
+        if (underlyingBalance < buffer || investedAssets > maxInvestedAmount) {
+            uint256 withdrawAmount;
+            if (underlyingBalance < buffer) {
+                withdrawAmount = buffer - underlyingBalance;
+                if (withdrawAmount > investedAssets) {
+                    withdrawAmount = investedAssets;
+                }
+            } else {
+                withdrawAmount = investedAssets - maxInvestedAmount;
+            }
+            if (withdrawAmount > maxRebalanceAmount) {
+                withdrawAmount = maxRebalanceAmount;
+            }
+            require(
+                withdrawAmount > 0 && withdrawAmount >= minRebalanceAmount, "ZkBobCompounding: insufficient rebalance"
+            );
+            investedAssetsAmount = investedAssets - withdrawAmount;
+            IERC4626(yieldAddress).withdraw(withdrawAmount, address(this), address(this));
+            emit Rebalance(yieldAddress, withdrawAmount, 0);
+        } else {
+            uint256 depositAmount = underlyingBalance - buffer;
+            if (investedAssets + depositAmount > maxInvestedAmount) {
+                depositAmount = maxInvestedAmount - investedAssets;
+            }
+            if (depositAmount > maxRebalanceAmount) {
+                depositAmount = maxRebalanceAmount;
+            }
+            require(
+                depositAmount > 0 && depositAmount >= minRebalanceAmount, "ZkBobCompounding: insufficient rebalance"
+            );
+            investedAssetsAmount = investedAssets + depositAmount;
+            IERC4626(yieldAddress).deposit(depositAmount, address(this));
+            emit Rebalance(yieldAddress, 0, depositAmount);
+        }
+    }
+
+    /**
+     * @dev Collects accumulated fees and generated yield for the specific collateral.
+     * Callable only by the contract owner / proxy admin / yield admin.
+     * @param minClaimAmount minimum amount of token to claim.
+     * @return Claimed amount.
+     */
+    function claim(uint256 minClaimAmount) external returns (uint256) {
+        (address yieldAddress, address operator, uint256 dust, address interestReceiver) =
+            (yieldParams.yield, yieldParams.yieldOperator, yieldParams.dust, yieldParams.interestReceiver);
+
+        require(yieldAddress != address(0), "ZkBobCompounding: yield not enabled");
+        require(operator == address(0) || operator == msg.sender || _isOwner(), "ZkBobCompounding: not authorized");
+
+        uint256 claimed = _claim(yieldAddress, interestReceiver, dust);
+
+        require(claimed > 0 && claimed >= minClaimAmount, "ZkBobCompounding: not enough to claim");
+
+        return claimed;
+    }
+
+    /**
+     * @dev Withdraws everything from the yield.
+     * @param targetAmount amount of token to withdraw.
+     * Callable only by the contract owner / proxy admin / yield admin.
+     */
+    function emergencyWithdraw(uint256 targetAmount) external onlyOwner {
```

Comment:
So, after some additional thinking and refactoring, I still don't see how this is gonna be used. How is this better than calling the following actions separately:
1. Call `updateYield` with `maxInvestedAmount = 0`
2. Call `rebalance` one or more time to withdraw all the funds according to their availability
3. Call `updateYield` to disable a yield completely and withdraw the remaining interest